app/com/m3/octoparts/util/OctoMetricsImpl.scala

```diff
@@ -0,0 +1,27 @@
+package com.m3.octoparts.util
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.kenshoo.play.metrics.MetricsImpl
+import play.api.{ Logger, Configuration }
+import play.api.inject.ApplicationLifecycle
+
+import scala.util.control.NonFatal
+
+/**
+ * During tests, we occasionally have cases where the asynchronous shutting down of an application means that
+ * the previous registry in the singleton has not been unregistered before starting the next test.
+ *
+ * This implementation of Metrics overrides onStart to perform onStop cleanups first before proceeding with
+ * other onStart duties
+ */
+class OctoMetricsImpl(lifecycle: ApplicationLifecycle, configuration: Configuration) extends MetricsImpl(lifecycle, configuration) {
+
+  override def onStart(): ObjectMapper = {
```

Comment:
Ah ok, so the catch here was more for paranoia (which I'm totally for when it comes to internal config issues).

Yeah, this plan sounds sounds (ugh).